lib/ansible/modules/crypto/entrust/ecs_certificate.py

```diff
@@ -0,0 +1,906 @@
+    type: int
+    sample: 380079
+serial_number:
+    description: The serial number of the issued certificate.
+    returned: success
+    type: int
+    sample: 1235262234164342
+cert_days:
+    description: The number of days the certificate remains valid.
+    returned: success
+    type: int
+    sample: 253
+cert_status:
+    description:
+        - The certificate status in ECS.
+        - 'Current possible values (which may be expanded in the future) are: C(ACTIVE), C(APPROVED), C(DEACTIVATED), C(DECLINED), C(EXPIRED), C(NA),
+          C(PENDING), C(PENDING_QUORUM), C(READY), C(REISSUED), C(REISSUING), C(RENEWED), C(RENEWING), C(REVOKED), C(SUSPENDED)'
+    returned: success
+    type: str
+    sample: ACTIVE
+cert_details:
+    description:
+        - The full response JSON from the Get Certificate call of the ECS API.
+        - 'While the response contents are guaranteed to be forwards compatible with new ECS API releases, Entrust recommends that you do not make any
+          playbooks take actions based on the content of this field. However it may be useful for debugging, logging, or auditing purposes.'
+    returned: success
+    type: dict
+
+'''
+
+from ansible.module_utils.ecs.api import (
+    ecs_client_argument_spec,
+    ECSClient,
+    RestOperationException,
+    SessionConfigurationException,
+)
+
+import datetime
+import json
+import os
+import time
+import traceback
+from distutils.version import LooseVersion
+
+from ansible.module_utils import crypto as crypto_utils
+from ansible.module_utils.basic import AnsibleModule, missing_required_lib
+from ansible.module_utils._text import to_native, to_bytes
+
+CRYPTOGRAPHY_IMP_ERR = None
+try:
+    import cryptography
+    CRYPTOGRAPHY_VERSION = LooseVersion(cryptography.__version__)
+except ImportError:
+    CRYPTOGRAPHY_IMP_ERR = traceback.format_exc()
+    CRYPTOGRAPHY_FOUND = False
+else:
+    CRYPTOGRAPHY_FOUND = True
+
+MINIMAL_CRYPTOGRAPHY_VERSION = '1.6'
+
+
+def calculate_cert_days(expires_after):
+    cert_days = 0
+    if expires_after:
+        expires_after_datetime = datetime.datetime.strptime(expires_after, '%Y-%m-%dT%H:%M:%SZ')
+        cert_days = (expires_after_datetime - datetime.datetime.now()).days
+    return cert_days
+
+
+# Populate the value of body[dict_param_name] with the JSON equivalent of
+# module parameter of param_name if that parameter is present, otherwise leave field
+# out of resulting dict
+def convert_module_param_to_json_bool(module, dict_param_name, param_name):
+    body = {}
+    if module.params.get(param_name, None) is not None:
+        if module.params[param_name]:
+            body[dict_param_name] = 'true'
+        else:
+            body[dict_param_name] = 'false'
+    return body
+
+
+class EcsCertificate(object):
+    '''
+    Entrust Certificate Services certificate class.
+    '''
+
+    def __init__(self, module):
+        self.path = module.params['path']
+        self.chain_path = module.params.get('chain_path')
```

Comment:
```suggestion
        self.chain_path = module.params['chain_path']
```
You can always use `[]` to access module options. If they're omitted, you'll get `None`. (`AnsibleModule` takes care of that.)